ios/engine/KMEI/KeymanEngine/Classes/KeymanPackage.swift

```diff
@@ -118,4 +119,24 @@ public class KeymanPackage {
                         }
                       })
   }
+
+  public func contains(_ resource: LanguageResource) -> Bool {
+    let matchesFound: [[LanguageResource]] = self.installableResourceSets.compactMap { set in
+      let setMatches: [LanguageResource] = set.compactMap { other in
+        if resource.id == other.id && resource.languageID == other.languageID {
+          return other
+        } else {
+          return nil
+        }
+      }
+
+      if setMatches.count > 0 {
+        return setMatches
+      } else {
+        return nil
+      }
+    }
+
+    return matchesFound.count > 0
```

Comment:
A reasonable question for `contains(_ resource: LanguageResource) -> Bool`:

As it stands, this will only confirm that a matching pairing is defined within the `KeymanPackage`.  It will _not_, however, use the package's defined `LanguageResource` pairing if being tested with a separately-constructed instance.  (See the new keyboard-installation unit test.)

We could change this implementation to return the match (thus, `-> LanguageResource?` instead of `-> Bool`), favoring the package's actual definition over a possibly-incomplete definition sourced from elsewhere in code.  Is this something we want to do, or should the caller be able to specify mild differences?  I feel like the answer is probably "yes", but I can see an argument for either direction.

Reply:
I'm not really understanding the question you are asking here. How are you intending this to be used? Is it just in unit tests?

Reply:
From actual kmp.json:

```
{
  "name": "Khmer Angkor",
  "id": "khmer_angkor",
  "version": "1.0.6",
  "oskFont": "Mondulkiri-R.ttf",
  "displayFont": "Mondulkiri-R.ttf",
  "languages": [
    {
      "name": "Central Khmer (Khmer, Cambodia)",
      "id": "km"
    }
  ]
}
```

Suppose that a KeymanEngine user had a definition equivalent to this as a hardcoded literal that they then passed in:

```
{
  "name": "Khmer Angkor",
  "id": "khmer_angkor",
  "version": "1.0.6",
  "languages": [
    {
      "name": "Central Khmer (Khmer, Cambodia)",
      "id": "km"
    }
  ]
}
```

In this case, should these two definitions "match"?  It's really a question more for #3207, and I ended up writing a method signature that accepts both variants.  (A second parameter indicates whether or not to be "strict".)

Reply:
Understood. But in what circumstances are you needing to do something other than an id-based match? Our design mandates that we identify packages based on id, not on content. 

So ... is this matching just for test cases, or do you intend to use it elsewhere? If it's for test cases, it belongs in test support code not in the core engine 😄 .

Reply:
It's something that must be considered due to the nature of the `KeymanEngine` framework - use in other apps may want either behavior.

I just pushed up a new commit that allows the behavior to be selected.  By default, the KMP's definition will win, but a `KeymanEngine` user may specify otherwise if desired.

Reply:
> It's something that must be considered due to the nature of the `KeymanEngine` framework - use in other apps may want either behavior.

Um ... no? The entire system is predicated on keyboards, packages and models keyed by id.

Reply:
Also, I'm not 100% on this yet, but it may be very useful for cloud resource migration.  

Reconstruction of the `KMPMetadata` from just a `LanguageResource` isn't perfect, and once a KMP is matched to a resource, it's hard to say at present which variant will be needed for further operations.  That's also why I shifted `contains -> Bool` to `findMatch -> LanguageResource?` - it's more helpful to sipmly return that match and let the caller decide whether or not to use it.

Reply:
> Um ... no? The entire system is predicated on keyboards, packages and models keyed by id.

I missed the part where I implied that it wasn't.  The issue is that it's _not also keyed_ on font, and debatably on version.  (We use 'version' for updates, but that's not always available directly on lexical-model definitions within kmp.json.)

Reply:
Sorted out in Standup!